core/trie2/trie.go

```diff
@@ -46,8 +46,18 @@ type Trie struct {
 	pendingUpdates int
 }
 
-// Creates a new trie
-func New(id trieutils.TrieID, height uint8, hashFn crypto.HashFn, nodeDB database.NodeDatabase) (*Trie, error) {
+// Creates a new trie, with the arguments:
+// - id: the trie id
+// - height: the height of the trie, 251 for contract and class trie
+// - hashFn: the hash function to use
+// - nodeDB: database interface, which provides methods to read and write trie nodes
+// - rootHash: the root hash of the trie (provided only for hash scheme of nodeDB, needed to properly recreate the trie)
+func New(id trieutils.TrieID,
+	height uint8,
+	hashFn crypto.HashFn,
+	nodeDB database.NodeDatabase,
+	rootHash *felt.Felt,
+) (*Trie, error) {
```

Comment:
I am not fully convinced here, I don't like the idea of having optional arguments. I would prefer to have two constructors one for hash db and another for path db. Is that what we are doing here or am I missing something?

Reply:
Yes exactly. Okey I agree, it isn't the cleanest. Added `NewWithRootHash` constructor, will use it later during the hashdb integration